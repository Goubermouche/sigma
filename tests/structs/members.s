i32 main() {
	struct vec2 {
		i32 x;
		i32 y;
		i32** arr;
	};

	vec2 instance;

	instance.x = 10;
	instance.y = 20;

	instance.arr = cast<i32**>(malloc(16));
	instance.arr[0] = cast<i32*>(malloc(4));
	instance.arr[1] = cast<i32*>(malloc(4));
	instance.arr[0][0] = 1;
	instance.arr[1][0] = 2;

	printf("x: %d\n", instance.x);
	printf("y: %d\n", instance.y);
	printf("arr: %d %d\n", instance.arr[0][0], instance.arr[1][0]);

	ret 0;
}
