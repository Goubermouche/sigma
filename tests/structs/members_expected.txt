x: 10
y: 20
arr: 1 2
